crc/services/workflow_service.py

```diff
@@ -180,13 +181,81 @@ def get_random_data_for_field(field, task):
     def __get_options(self):
         pass
 
-
     @staticmethod
     def _random_string(string_length=10):
         """Generate a random string of fixed length """
         letters = string.ascii_lowercase
         return ''.join(random.choice(letters) for i in range(string_length))
 
+    @staticmethod
+    def processor_to_workflow_api(processor: WorkflowProcessor, next_task=None):
+        """Returns an API model representing the state of the current workflow, if requested, and
+        possible, next_task is set to the current_task."""
+
+        nav_dict = processor.bpmn_workflow.get_nav_list()
+        navigation = []
+        for nav_item in nav_dict:
+            spiff_task = processor.bpmn_workflow.get_task(nav_item['task_id'])
+            if 'description' in nav_item:
+                nav_item['title'] = nav_item.pop('description')
+                # fixme: duplicate code from the workflow_service. Should only do this in one place.
+                if ' ' in nav_item['title']:
+                    nav_item['title'] = nav_item['title'].partition(' ')[2]
+            else:
+                nav_item['title'] = ""
+            if spiff_task:
+                nav_item['task'] = WorkflowService.spiff_task_to_api_task(spiff_task, add_docs_and_forms=False)
+                nav_item['title'] = nav_item['task'].title  # Prefer the task title.
+            else:
+                nav_item['task'] = None
+            if not 'is_decision' in nav_item:
+                nav_item['is_decision'] = False
+
+            navigation.append(NavigationItem(**nav_item))
+            NavigationItemSchema().dump(nav_item)
+
+        spec = db.session.query(WorkflowSpecModel).filter_by(id=processor.workflow_spec_id).first()
+        workflow_api = WorkflowApi(
+            id=processor.get_workflow_id(),
+            status=processor.get_status(),
+            next_task=None,
+            navigation=navigation,
+            workflow_spec_id=processor.workflow_spec_id,
+            spec_version=processor.get_version_string(),
+            is_latest_spec=processor.is_latest_spec,
+            total_tasks=len(navigation),
+            completed_tasks=processor.workflow_model.completed_tasks,
+            last_updated=processor.workflow_model.last_updated,
+            title=spec.display_name
+        )
+        if not next_task:  # The Next Task can be requested to be a certain task, useful for parallel tasks.
+            # This may or may not work, sometimes there is no next task to complete.
+            next_task = processor.next_task()
+        if next_task:
+            workflow_api.next_task = WorkflowService.spiff_task_to_api_task(next_task, add_docs_and_forms=True)
+
+        return workflow_api
+
+    @staticmethod
+    def get_previously_submitted_data(workflow_id, task):
```

Comment:
Where do we actually call this method? I didn't see it in the pull request - perhaps I'm missing something?